### DISCUSSION TITLE
how I can put retryDelaySeconds in DO_WHILE task

### DISCUSSION THREAD

**Discussion by bhawani021** [Netflix/conductor#2509] [Q&A]
```
{
  "name": "test_workflow",
  "description": "test workflow",
  "version": 1,
  "tasks": [
    {
      "name": "loop_task",
      "taskReferenceName": "loop",
      "type": "DO_WHILE",
      "inputParameters": {
        "value": 5
      },
      "loopCondition": "if ( ($.LoopTask['iteration'] < $.value ) ) { false; } else { true; }",
      "loopOver": [
        {
          "name": "test_task_test",
          "taskReferenceName": "ref1",
          "inputParameters": {
            "number": 1
          },
          "type": "SIMPLE",
          "retryDelaySeconds": 30
        }
      ]

    }
  ],
  "inputParameters": [],
  "outputParameters": {},
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "bhawani.shanker@gmail.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {}
}

```

I tried to put           "retryDelaySeconds": 30
so my task test_task_test execute after every 30 secs but it's not working.
Any suggestion?

**Comment by SukhadB** [comment]
Hi, I am trying to use DO_WHILE in my workflow, however it fails with "Invalid task specified. Cannot find task by name loop_task in the task definitions", loop_task is the taskRefName. With the above workflow definition are you able to execute the DO_WHILE task? Can you help me with the Workflow definition with DO_WHILE task?

**Comment by aravindanr** [comment]
@bhawani021 `retryDelaySeconds` is only supported in the task definition. For this case, you can add the field to the task definition of `test_task_test`.

**Comment by aravindanr** [comment]
> Hi, I am trying to use DO_WHILE in my workflow, however it fails with "Invalid task specified. Cannot find task by name loop_task in the task definitions", loop_task is the taskRefName. With the above workflow definition are you able to execute the DO_WHILE task? Can you help me with the Workflow definition with DO_WHILE task?

It would help if you can post the workflow definition. Based on the error message, it looks like the workflow is using a `SIMPLE` task named `loop_task` for which there is no task definition defined.
